lib/Doctrine/KeyValueStore/Storage/MongoDbStorage.php

```diff
@@ -112,55 +79,63 @@ public function requiresCompositePrimaryKeys()
      */
     public function insert($storageName, $key, array $data)
     {
-        $this->initialize();
-
-        $value = [
-            'key'   => $key,
-            'value' => $data,
-        ];
-
-        $this->collection->insert($value);
+        $this->client
+            ->selectDatabase($this->database)
+            ->selectCollection($storageName)
+            ->insertOne([
+                'key'   => $key,
+                'value' => $data,
+            ]);
     }
 
     /**
      * {@inheritDoc}
      */
     public function update($storageName, $key, array $data)
     {
-        $this->initialize();
-
-        $value = [
-            'key'   => $key,
-            'value' => $data,
-        ];
-
-        $this->collection->update(['key' => $key], $value);
+        $this->client
+            ->selectDatabase($this->database)
+            ->selectCollection($storageName)
+            ->updateOne([
+                'key'   => $key,
+            ], [
+                '$set' => [
```

Comment:
Since you're replacing the entire document anyways, you may as well use `replaceOne` instead of `updateOne`.